### PR TITLE
Developed BRS for control 'Azure_Subscription_Configure_Conditional_A…

### PR DESCRIPTION
**Pull request by Aditi-sha**
…ccess_for_PIM'

### CODE REVIEW

**Review by VamseeInala — APPROVED** [review]

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 [hunk 1]
```diff
@@ -0,0 +1,723 @@
...
+        .EXAMPLE
+        PS> Disable-ConditionalAccessPolicyForPIM -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -RoleName 'User Access Administrator'.
+
+        .LINK
+        None
+    #>
+
+    param (
+        [String]
+        [Parameter(ParameterSetName = "WetRun", Mandatory = $true, HelpMessage="Specifies the ID of the Subscription to be remediated")]
```
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:517]: Should this be roll back?

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+        [Parameter(ParameterSetName = "CustomConfig", Mandatory = $true, HelpMessage="Specifies the ID of the Subscription to be remediated")]
+        $SubscriptionId,
+
+        [Switch]
+        [Parameter(ParameterSetName = "WetRun", HelpMessage="Specifies validation of prerequisites for the command")]
+        [Parameter(ParameterSetName = "CustomConfig", HelpMessage="Specifies validation of prerequisites for the command")]
+        $PerformPreReqCheck,
+
+        [String]
+        [Parameter(ParameterSetName = "CustomConfig", Mandatory = $false, HelpMessage="Specifies the role name on which CA policy needs to be configured")]
```
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:527]: Should this be 'disabled'?

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+        [ValidateSet("Owner", "Contributor", "User Access Administrator")]
+        $RoleName,
+
+        [String]
+        [Parameter(ParameterSetName = "WetRun", Mandatory = $true, HelpMessage="Specifies the path to the file to be used as input for the remediation")]
```
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:532]: Should this be 'disabled'?

**Review by gvaradarajan-msft — CHANGES_REQUESTED** [review]

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 (hunk 2 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:527]: Done

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 (hunk 3 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:532]: Done.

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 (hunk 1 above)
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:517]: Done.

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 [hunk 4]
```diff
@@ -0,0 +1,723 @@
...
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "`n[Step 4 of 5] Backing up Conditional Access policy details..."
+
+    # Backing up App Services details.
```
[the remark this answers is not in the document]
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:363]: Done.

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 [hunk 5]
```diff
@@ -0,0 +1,723 @@
...
+    {
+        Write-Host "WARNING: This script can only be run by an [$($requiredRoleDefinitionName -join ", ")]. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+        return;
+    }
+    else
+    {
+        Write-Host "Current user [$($context.Account.Id)] has the required permission for subscription [$($SubscriptionId)].`n" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # No file path provided as input to the script. Fetch all Kubernetes Services in the Subscription.
```
[the remark this answers is not in the document]
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:289]: Done.

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 [hunk 6]
```diff
@@ -0,0 +1,723 @@
...
+
+        # Write this to a file.
+        $remediationSummaryFile = "$($backupFolderPath)\ConfiguredCAPolicy.csv"
+        $remediationSummary | Export-CSV -Path $remediationSummaryFile -NoTypeInformation
+        Write-Host "This information has been saved to $($remediationSummaryFile)"
+        Write-Host "Use this file for any roll back that may be required." -ForegroundColor $([Constants]::MessageType.Info)
+    }
+    else
+    {
+        Write-Host $([Constants]::DoubleDashLine)
```
[the remark this answers is not in the document]
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:466]: Done.

File: Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1 [hunk 7]
```diff
@@ -0,0 +1,723 @@
...
+    else
+    {
+        Write-Host "Current user [$($context.Account.Id)] has the required permission for subscription [$($SubscriptionId)].`n" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # No file path provided as input to the script. Fetch all Kubernetes Services in the Subscription.
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "`n[Step 3 of 5] Fetching currently configured Conditional Access (CA) policy for subscription [$($SubscriptionId)]..."
```
[the remark this answers is not in the document]
Comment (Aditi-sha) [reply Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1:293]: In if block, we are fetching configurations from subscription whereas in else block, we are fetching configurations from given CSV and hence added two different logs.

**Review by gvaradarajan-msft — APPROVED** [review]

**Pull request merged**